src/source/Signaling/Client.c

```diff
@@ -16,7 +40,28 @@ STATUS createSignalingClientSync(PSignalingClientInfo pClientInfo, PChannelInfo
     MEMSET(&signalingClientInfoInternal, 0x00, SIZEOF(signalingClientInfoInternal));
     signalingClientInfoInternal.signalingClientInfo = *pClientInfo;
 
-    CHK_STATUS(createSignalingSync(&signalingClientInfoInternal, pChannelInfo, pCallbacks, pCredentialProvider, &pSignalingClient));
+    CHK_STATUS(validateSignalingClientRetryStrategy(pClientInfo));
```

Comment:
Same here. Do we want to pass the entire struct to validate retry strategy? No harm in passing it though.

Reply:
Yeah same reason. Need to make sure `pClientInfo->signalingClientCreationMaxRetryCount` is at least 1.